pydis_site/apps/api/tests/migrations/test_active_infraction_migration.py

```diff
@@ -0,0 +1,545 @@
+                    infractions=(
+                        {'type': 'ban', 'active': True, 'hidden': True},
+                        {'type': 'mute', 'active': True, 'hidden': True},
+                        {'type': 'superstar', 'active': True, 'hidden': True},
+                        {'type': 'watch', 'active': True, 'hidden': True},
+                    ),
+                    infraction_model=cls.infraction_model,
+                    user_model=cls.user_model,
+                )
+            }
+        )
+
+        # User #12: Multiple active bans, active mutes, active superstars
+        cls.created_infractions.update(
+            {
+                12: InfractionFactory.create(
+                    actor=cls.user_moderator,
+                    infractions=(
+                        {'type': 'ban', 'active': True, 'hidden': True},
+                        {'type': 'ban', 'active': True, 'hidden': True},
+                        {'type': 'ban', 'active': True, 'hidden': True},
+                        {'type': 'mute', 'active': True, 'hidden': True},
+                        {'type': 'mute', 'active': True, 'hidden': True},
+                        {'type': 'mute', 'active': True, 'hidden': True},
+                        {'type': 'superstar', 'active': True, 'hidden': True},
+                        {'type': 'superstar', 'active': True, 'hidden': True},
+                        {'type': 'superstar', 'active': True, 'hidden': True},
+                        {'type': 'watch', 'active': True, 'hidden': True},
+                        {'type': 'watch', 'active': True, 'hidden': True},
+                        {'type': 'watch', 'active': True, 'hidden': True},
+                    ),
+                    infraction_model=cls.infraction_model,
+                    user_model=cls.user_model,
+                )
+            }
+        )
+
+    def test_all_never_active_types_became_inactive(self):
+        """Are all infractions of a non-active type inactive after the migration?"""
+        inactive_type_query = Q(type="note") | Q(type="warning") | Q(type="kick")
+        self.assertFalse(
+            self.infraction_model.objects.filter(inactive_type_query, active=True).exists()
+        )
+
+    def test_migration_left_clean_user_without_infractions(self):
+        """Do users without infractions have no infractions after the migration?"""
+        user_id, infraction_history = self.created_infractions[1]
+        self.assertFalse(
+            self.infraction_model.objects.filter(user__id=user_id).exists()
+        )
+
+    def test_migration_left_user_with_inactive_note_untouched(self):
+        """Did the migration leave users with only an inactive note untouched?"""
+        user_id, infraction_history = self.created_infractions[2]
+        inactive_note = infraction_history[0]
+        self.assertTrue(
+            self.infraction_model.objects.filter(**model_to_dict(inactive_note)).exists()
```

Comment:
Do all these filter by infraction ID too? Or is that not necessary?

Reply:
Yes, they do. The infraction object returned by the InfractionFactory will include the `id` (and all the other fields of the created `Infraction` object).